core/crypto/vrf/p256/p256.go

```diff
@@ -213,6 +217,22 @@ func (pk *PublicKey) ProofToHash(m, proof []byte) (index [32]byte, err error) {
 	return sha256.Sum256(vrf), nil
 }
 
+// NewFromWrappedKey creates a signer object from an encrypted private key.
+func NewFromWrappedKey(ctx context.Context, wrapped proto.Message) (vrf.PrivateKey, error) {
+	// Unwrap.
+	signer, err := keys.NewSigner(ctx, wrapped)
+	if err != nil {
+		return nil, err
+	}
+
+	switch key := signer.(type) {
```

Comment:
Instead of testing the type of the signer, I recommend testing the type of the public key instead (see [SignatureAlgorithm(crypto.PublicKey)](https://github.com/google/trillian/blob/f415147c1e56ab0b7289a5e0b16dd6531bf4cf90/crypto/signatures.go#L27)). It's possible that some key proto handlers will not return a raw key, in which case this code will fail. For example, imagine a key proto handler that communicates with a remote key store to perform signing operations. The public key is much more likely to be a raw key though.

Reply:
How would you recommend this code handle a raw key? At the end of the day I need an `ecdsa.PrivateKey` in order for the rest of this library to function correctly.  Is there a conversion function I might need to call? Would a `switch` be needed?

Reply:
Ah I see, you need access to `key.D`.  In that case, you can leave this as it is. For the kind of signer I talked about, it quite possibly won't allow access to the raw key (e.g. a HSM with non-extractable key) so you'd just have to return an error in that case.

Reply:
It might be worth adding a little more documentation to this func to state that `wrapped` must identify a raw ECDSA private key, or something to that effect (I'm not sure what the best terms are to use).

Reply:
ok, that makes sense.  Yes, this particular library is doing some pretty new / not (yet) standardized crypto on the Elliptic Curve itself.  Much fancier than `crypto.Signer` :-)  I'll add a note.